app/controllers/redirections_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RedirectionsController < ApplicationController
+  def next
+    redirection = Redirection.find_by(slug: params[:slug])
+
+    redirect_to redirection.next.url
+  end
+
+  def previous
```

Comment:
I don't love not using RESTful actions, but hey: Proof Of Concept! Proof Of Concept! WOOOOO